packages/core/src/tools/error.spec.ts

```diff
@@ -107,3 +190,48 @@ describe('createHandlingStack', () => {
   at userCallOne @ (.*)`)
   })
 })
+
+describe('flattenErrorCauses', () => {
+  it('should return undefined if no cause found', () => {
+    const error = new Error('foo') as ErrorWithCause
+    const errorCauses = flattenErrorCauses(error, ErrorSource.LOGGER)
+    expect(errorCauses).toEqual(undefined)
+  })
+
+  it('should stop recursive loop if cause is not of type Error', () => {
+    const error = new Error('foo') as ErrorWithCause
+    const nestedError = new Error('bar')
+    const deepNestedError = { biz: 'buz', cause: new Error('boo') }
+    error.cause = nestedError
+    // @ts-ignore: want to test we can handle non error cases
+    nestedError.cause = deepNestedError
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about having the non-error cause in the first error?
it would simplify the current test and I don't think that testing that it continues to work at a deeper level bring much more value.
wdyt?